Add GPT-OSS models + more (#12)
groq/groq-changelog: CHANGELOG.md

# Groq Changelog

## 2025-08-05 (Python SDK v0.31.0, TypeScript SDK v0.30.0)

### [ADDED] OpenAI GPT-OSS 20B & OpenAI GPT-OSS 120B

> [GPT-OSS 20B](https://console.groq.com/docs/model/openai/gpt-oss-20b) and [GPT-OSS 120B](https://console.groq.com/docs/model/openai/gpt-oss-120b) are OpenAI's open-source state-of-the-art Mixture-of-Experts (MoE) language models with 20 billion and 120 billion total parameters. It has [reasoning](https://console.groq.com/docs/reasoning) capabilities, built in [browser search](https://console.groq.com/docs/browser-search) and [code execution](https://console.groq.com/docs/code-execution), and support for [structured outputs](https://console.groq.com/docs/structured-outputs). 

Reviewer: ```suggestion
[GPT-OSS 20B](https://console.groq.com/docs/model/openai/gpt-oss-20b) and [GPT-OSS 120B](https://console.groq.com/docs/model/openai/gpt-oss-120b) are OpenAI's open-source state-of-the-art Mixture-of-Experts (MoE) language models that perform as well as their o4-mini and o3-mini models. They have [reasoning](https://console.groq.com/docs/reasoning) capabilities, built-in [browser search](https://console.groq.com/docs/browser-search) and [code execution](https://console.groq.com/docs/code-execution), and support for [structured outputs](https://console.groq.com/docs/structured-outputs). 
```

    "messages": [
      {
        "role": "user",
        "content": "Explain why fast inference is critical for reasoning models"
      }
    ]
  }'
```

### [ADDED] Responses API (Beta)

[Groq's Responses API](https://console.groq.com/docs/responses-api) is fully compatible with OpenAI's Responses API, making it easy to integrate advanced conversational AI capabilities into your applications. The [Responses API](https://console.groq.com/docs/responses-api) supports both text and image inputs while producing text outputs, stateful conversations, and function calling to connect with external systems.

This feature is in beta right now - please let us know your feedback on our [Community Forum](https://community.groq.com/)!

**Example Usage:**
```curl
curl https://api.groq.com/openai/v1/responses \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer $GROQ_API_KEY" \
  -d '{
    "model": "llama-3.3-70b-versatile",
    "input": "Tell me a fun fact about the moon in one sentence."
  }'
```

### [CHANGED] Python SDK v0.31.0, TypeScript SDK v0.30.0

The Python SDK has been updated to v0.30.0 and the Typescript SDK has been updated to v0.27.0.

**Key Changes:**

>  - Added support for `high`, `medium`, and `low` options for `reasoning_effort` when using GPT-OSS models to control its reasoning amount. [Learn more about how to use these options to control reasoning tokens.](https://console.groq.com/docs/reasoning#reasoning-effort)

Reviewer: ```suggestion
 - Added support for `high`, `medium`, and `low` options for `reasoning_effort` when using GPT-OSS models to control their reasoning output. [Learn more about how to use these options to control reasoning tokens.](https://console.groq.com/docs/reasoning#reasoning-effort)
```
